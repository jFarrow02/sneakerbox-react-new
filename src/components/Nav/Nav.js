import React from 'react';
import './Nav.scss';
import {Link} from 'react-router-dom';
import {linkConfig} from '../../data/config';

const Nav = ()=>{

    return(
        <nav className='Nav'>
            Nav goes here
            {
                linkConfig.map((conf, idx)=>{
                    return(
                        <Link key={`nav-${idx}`} to={conf.url}>{conf.text}</Link>
                    );
                })
            }
        </nav>
    );
};

export default Nav;