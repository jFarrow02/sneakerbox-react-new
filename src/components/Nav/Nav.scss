@import '../../styles/variables.scss';

$paddingVert: 20px;
$paddingHoriz: calc(#{$paddingVert} / 2);


.Nav{
    grid-column: 1 /span 1;
    grid-row: 2 /span 1;
    a{
        display: block;
        text-decoration: none;
        background-color: $copper;
        color: $offWhite;
        padding-top: $paddingVert;
        padding-left: $paddingHoriz;
        padding-bottom: $paddingVert;
    }
    a:hover{
        background-color: $deepRed;
    }
}