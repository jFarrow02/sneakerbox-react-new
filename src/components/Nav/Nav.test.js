import React from 'react';
import Nav from './Nav';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<BrowserRouter><Nav/></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});