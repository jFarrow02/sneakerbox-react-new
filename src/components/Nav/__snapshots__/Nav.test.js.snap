// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<nav
  className="Nav"
>
  <a
    href="/home"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    href="/products/men"
    onClick={[Function]}
  >
    Men
  </a>
  <a
    href="/products/women"
    onClick={[Function]}
  >
    Women
  </a>
  <a
    href="/products/kids"
    onClick={[Function]}
  >
    Kids
  </a>
  <a
    href="/products/sale"
    onClick={[Function]}
  >
    Sale!
  </a>
  <a
    href="/myaccount"
    onClick={[Function]}
  >
    My Account
  </a>
</nav>
`;
